# Provenance: PAR-specific ABX detection fix by yukino1111 against TeamWin
# recovery. Upstream file copyright and license apply.
diff --git a/partitionmanager.cpp b/partitionmanager.cpp
index eff1034..22bc0d0 100755
--- a/partitionmanager.cpp
+++ b/partitionmanager.cpp
@@ -1737,9 +1737,9 @@ void TWPartitionManager::Parse_Users() {
 
 			// Attempt to get name of user. Fallback to user ID if this fails.
 			std::string path = "/data/system/users/" + to_string(userId) + ".xml";
-			if ((atoi(TWFunc::System_Property_Get("ro.build.version.sdk").c_str()) > 30) && TWFunc::Path_Exists(path)) {
-				if(!TWFunc::Check_Xml_Format(path.c_str()))
-					user.userName = to_string(userId);
+			if (TWFunc::Path_Exists(path) &&
+					!TWFunc::Check_Xml_Format(path.c_str())) {
+				user.userName = to_string(userId);
 			}
 			else {
 				char* userFile = PageManager::LoadFileToBuffer(path, NULL);
@@ -2894,11 +2894,10 @@ bool TWPartitionManager::Decrypt_Adopted() {
 	// In Android 12 xml format changed. Previously it was human-readable format with xml tags
 	// now it's ABX (Android Binary Xml). Sadly, rapidxml can't parse it, so check xml format firstly
 	std::string path = "/data/system/storage.xml";
-	if ((atoi(TWFunc::System_Property_Get("ro.build.version.sdk").c_str()) > 30) && TWFunc::Path_Exists(path))
-		if(!TWFunc::Check_Xml_Format(path.c_str())) {
-			LOGINFO("Android 12+: storage.xml is in ABX format. Skipping adopted storage decryption\n");
-			return false;
-		}
+	if (TWFunc::Path_Exists(path) && !TWFunc::Check_Xml_Format(path.c_str())) {
+		LOGINFO("Android 12+: storage.xml is in ABX format. Skipping adopted storage decryption\n");
+		return false;
+	}
 
 	LOGINFO("Decrypt adopted storage starting\n");
 	char* xmlFile = PageManager::LoadFileToBuffer("/data/system/storage.xml", NULL);
